tests/functional/tests/engine/test_io_zero_size.py

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+from ctypes import c_int
+
+from pyocf.types.cache import Cache
+from pyocf.types.core import Core
+from pyocf.types.volume import Volume
+from pyocf.types.data import Data
+from pyocf.types.io import IoDir
+from pyocf.utils import Size
+from pyocf.types.shared import OcfCompletion
+
+
+def test_io_zero_size(pyocf_ctx):
+    cache_device = Volume(Size.from_MiB(30))
+    core_device = Volume(Size.from_MiB(60))
+
+    cache = Cache.start_on_device(cache_device)
+    core = Core.using_device(core_device)
+
+    cache.add_core(core)
+
+    cache_device.reset_stats()
+    core_device.reset_stats()
+
+    io = core.new_core_io(cache.get_default_queue(), Size.from_sector(1).B,
+                          0, IoDir.WRITE, 0, 0)
+    io.set_data(0)
+
+    cmpl = OcfCompletion([("err", c_int)])
+    io.callback = cmpl.callback
+    io.submit()
+    cmpl.wait()
+
+    assert cmpl.results["err"] != 0
+    assert cache_device.get_stats()[IoDir.WRITE] == 0
+    stats = cache.get_stats()
+    assert stats["req"]["wr_full_misses"]["value"] == 0
```

Comment:
I don't think that misses should be checked here - it's only obscuring the more important asserts